mlir/tools/mlir-tblgen/AttrOrTypeFormatGen.cpp

```diff
@@ -424,9 +424,11 @@ void DefFormat::genVariableParser(ParameterElement *el, FmtContext &ctx,
         Dialect dialect(dialectInit->getDef());
         auto cppNamespace = dialect.getCppNamespace();
         std::string name = dialect.getCppClassName();
-        dialectLoading = ("\nodsParser.getContext()->getOrLoadDialect<" +
-                          cppNamespace + "::" + name + ">();")
-                             .str();
+        if (name != "BuiltinDialect" || cppNamespace != "::mlir") {
```

Comment:
I don't quite understand the second part of the condition. Why don't we want to load the dialect if its namespace starts with `::mlir`? These shouldn't be anyhow special.

Reply:
Overall this condition is the logical negation of `name == "BuiltinDialect" && namespace == "::mlir"`. For some unholy reason, someone might make an out-of-tree dialect called `BuiltinDialect` in a different namespace. In that case, we would still want to load it.

Reply:
I could also alternatively have getOrLoadDialect be aware of `BuiltinDialect` and handle it specially.